File: Leetcode/Array/Easy/MaxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit, minBuy = 0, prices[0]

        for price in prices[1:]:
            minBuy = minBuy if minBuy < price else price
            maxProfit = maxProfit if maxProfit > price - minBuy else price - minBuy

        return maxProfit

File: Leetcode/Array/Easy/MaxProfit.java
package Leetcode.Array.Easy;

public class MaxProfit {
    public int solution(int[] prices) {
        int maxProfit = 0;
        int minBuy = prices[0];

        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < minBuy) {
                minBuy = prices[i];
            }

            if (prices[i] - minBuy > maxProfit) {
                maxProfit = prices[i] - minBuy;
            }
        }

        return maxProfit;
    }

    public static void main(String[] args) {
        MaxProfit mp = new MaxProfit();
        int[] input = new int[]{7, 1, 5, 3, 6, 4};
        System.out.println((mp.solution(input)));
    }
}
